src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,123 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    /**
+     * Reads the input given and returns the appropriate Command to execute.
+     *
+     * @param input User input string to parse.
+     * @return Command object.
+     * @throws DukeException If input is not recognised or does not follow format.
+     */
+    public static Command parse(String input) throws DukeException {
+        String[] wordArray = input.split(" ");
+        String type = "";
+        String description = "";
+        LocalDateTime time = null;
+
+        if (wordArray.length == 0) {
+            throw new DukeException("Please enter a command!");
+        } else {
+            type = wordArray[0];
+        }
+
+        String timeString = "";
+        int pointer = 1;
+        while (pointer < wordArray.length
+                && !wordArray[pointer].equals("/by")
+                && !wordArray[pointer].equals("/at")) {
+            description += wordArray[pointer] + " ";
+            pointer++;
+        }
+        pointer++;
+        description = description.trim();   // delete the last space
+
+        while (pointer < wordArray.length) {
+            timeString += wordArray[pointer] + " ";
+            pointer++;
+        }
+        timeString = timeString.trim();   // delete the last space
+
+        try {
+            time = LocalDateTime.parse(timeString, FORMATTER);
+        } catch (DateTimeParseException e) {
+            // Does nothing
+        }
+
+        switch (type) {
+        case ("list"):
+            description += timeString;
+            if (description.equals("")) {
+                return new ListCommand();
+            } else {
+                throw new DukeException("Command 'list' does not need additional arguments");
+            }
```

Comment:
I notice you did not use `break;`. Perhaps you can write comments explaining why it is not necessary to put it.